docs/eventing/samples/writing-event-source-easy-way/README.md

```diff
@@ -0,0 +1,305 @@
+```
+to employ structured mode.
+
+However, binary mode should be used in most of the cases as:
+- It is faster in terms of serialization and deserialization
+- It works better with cloudevents-aware proxies (like Knative Channels) can simply check the header instead of parsing the payload
+
+## Making use of SinkBinding
+
+`SinkBinding` is a more powerful way of making any Kubernetes resource an event source.
+
+`ContainerSource` will create the container for your event source's image and it will be `ContainerSource` responsibility to manage the container.
+
+`SinkBinding` though, will not create any containers. It will inject the sink information to the already existing Kubernetes resources.
+This is a more flexible approach as you can use any Kubernetes `PodSpecable` as an event source, such as `Deployment`, `Job`, `Knative Service`, `DaemonSet` etc.
+
+We don't need any code changes in our source for making it work with `SinkBinding`.
+
+Create the event display as in the section before:
+```bash
+cat <<EOS |kubectl apply -f -
+---
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+spec:
+  template:
+    spec:
+      containers:
+        - image: docker run --name event_display --rm -p 8080:8080 docker.io/aliok/event_display-864884f202126ec3150c5fcef437d90c@sha256:93cb4dcda8fee80a1f68662ae6bf20301471b046ede628f3c3f94f39752fbe08
+EOS
+```
+
+Create a Kubernetes deployment that runs the event source:
+```bash 
+cat <<EOS |kubectl apply -f -
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: node-heartbeats-deployment
+  labels:
+    app: node-heartbeats
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: node-heartbeats
+  template:
+    metadata:
+      labels:
+        app: node-heartbeats
+    spec:
+      containers:
+      - name: node-heartbeats
+        image: path/to/image/registry/node-knative-heartbeat-source:v1
+        ports:
+        - containerPort: 8080
+EOS
+```
+
+As the `SinkBinding` is not created yet, `K_SINK` environment variable is not yet injected and the event source will complain about that.
+
+```
+$ kubectl logs node-heartbeats-deployment-9ffbb644b-llkzk
+
+Sink URL is undefined
+Emitting event #1
+Error during event post
+TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Received type undefined
+```
+
+Create the `SinkBinding`:
+```bash
+cat <<EOS |kubectl apply -f -
+---
+apiVersion: sources.knative.dev/v1alpha1
+kind: SinkBinding
+metadata:
+  name: bind-node-heartbeat
+spec:
+  subject:
+    apiVersion: apps/v1
+    kind: Deployment
+    selector:
+      matchLabels:
+        app: node-heartbeats
+  sink:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: event-display
+EOS
+```
+
+
+You will see the pods are recreated and this time the `K_SINK` environment variable is injected.
+
+Also note that since the `replicas` is set to 2, there will be 2 pods that are posting events to the sink. 
```

Comment:
<!--whitespace-bot-->
Remove trailing whitespace:
```suggestion
Also note that since the `replicas` is set to 2, there will be 2 pods that are posting events to the sink.
```